ui/app/mirrors/edit/[mirrorId]/cdc.tsx

```diff
@@ -87,35 +95,112 @@ function summarizeTableClone(clone: QRepMirrorStatus): TableCloneSummary {
 type SnapshotStatusProps = {
   status: SnapshotStatus;
 };
+
+const ROWS_PER_PAGE = 10;
 export const SnapshotStatusTable = ({ status }: SnapshotStatusProps) => {
-  const [snapshotRows, setSnapshotRows] = useState(
-    status.clones.map(summarizeTableClone)
+  const allRows = status.clones.map(summarizeTableClone);
+  const [sortField, setSortField] = useState<
+    'cloneStartTime' | 'avgTimePerPartition'
+  >('cloneStartTime');
+  const [currentPage, setCurrentPage] = useState(1);
+  const totalPages = Math.ceil(allRows.length / ROWS_PER_PAGE);
+
+  const startRow = (currentPage - 1) * ROWS_PER_PAGE;
+  const endRow = startRow + ROWS_PER_PAGE;
+  const [displayedRows, setDisplayedRows] = useState(() =>
+    allRows.slice(startRow, endRow)
+  );
+
+  const handlePrevPage = () => {
+    if (currentPage > 1) {
+      setCurrentPage(currentPage - 1);
+      const newStartRow = (currentPage - 2) * ROWS_PER_PAGE;
+      const newEndRow = newStartRow + ROWS_PER_PAGE;
+      setDisplayedRows(allRows.slice(newStartRow, newEndRow));
+    }
+  };
+
+  const handleNextPage = () => {
+    if (currentPage < totalPages) {
+      setCurrentPage(currentPage + 1);
+      const newStartRow = currentPage * ROWS_PER_PAGE;
+      const newEndRow = newStartRow + ROWS_PER_PAGE;
+      setDisplayedRows(allRows.slice(newStartRow, newEndRow));
+    }
+  };
```

Comment:
the logic to setDisplayedRows can be extracted into another useffect which is dependent on current page, when current page changes, it will update displayedRows

Reply:
you can also consider using a pagination hook for this https://github.com/erictooth/react-use-pagination

Reply:
`setDisplayedRows` has now been removed as we are now using `useMemo` here and for `SyncStatusTable`